serde/src/tagger.rs

```diff
@@ -0,0 +1,23 @@
+/// Provides the correct tag for a given format.
+pub trait Tagger {
```

Comment:
I wonder if this could somehow be made generic over [`Visitor`](https://serde-rs.github.io/serde/serde/de/trait.Visitor.html), so the implementor simply calls a method on the passed `Visitor`.